### PR TITLE
Arc::from_raw_slice

### LINKED ISSUE #56: Arc::from_raw requires Sized
`Arc::into_raw` does not require Sized, but `Arc::from_raw` does. Must be a mistake.

Edit: I missed todo

https://github.com/Manishearth/triomphe/blob/0c63099082ba155885c304b3b42e424b676da8ee/src/arc.rs#L81

However, for now `Arc` can have an operation like `Arc::from_raw_slice`. Practically many unsized types are just slices.

### PR DESCRIPTION
**Pull request by stepancheg**
Fixes #56

### CODE REVIEW

File: src/arc.rs [hunk 1]
```diff
@@ -144,6 +144,24 @@ impl<T> Arc<T> {
...
+    /// Reconstruct the `Arc<[T]>` from a raw pointer obtained from `into_raw()`.
+    ///
+    /// [`Arc::from_raw`] should accept unsized types, but this is not trivial to do correctly
+    /// until the feature [`pointer_bytes_offsets`](https://github.com/rust-lang/rust/issues/96283)
+    /// is stabilized. This is stopgap solution for slices.
+    pub unsafe fn from_raw_slice(ptr: *const [T]) -> Self {
+        let len = (*ptr).len();
+        // Assuming the offset of `T` in `ArcInner<T>` is the same
+        // as as offset of `[T]` in `ArcInner<[T]>`.
+        let arc_inner_ptr = (ptr as *const u8).sub(offset_of!(ArcInner<T>, data));
```
Comment (Manishearth) [inline src/arc.rs:157]: question: is it not possible to just use `ArcInner<[T]>`?

Comment (Manishearth) [reply src/arc.rs:157]: but the assumption is probably correct, at least due to CoerceUnsize behavior

**Review by Manishearth — APPROVED** [review]

File: src/arc.rs (hunk 1 above)
Comment (stepancheg) [reply src/arc.rs:158]: > question: is it not possible to just use ArcInner<[T]>?

`offset_of!` macro doesn't work with unsized types.

**Comment by stepancheg** [comment]
Rebased on master.

**Review by Manishearth — APPROVED** [review]

**Comment by stepancheg** [comment]
Added a comment.

**Comment by Manishearth** [comment]
Thanks!

**Pull request merged**